src/utils/dockerDirManager.ts

```diff
@@ -270,41 +257,49 @@ export default class DockerDirManager {
   }
 
   /**
-   * Ensures that the state of everything under the docker CLI config directory
-   * is correct.
+   * Ensures that the Rancher Desktop context file exists, and that the docker context
+   * is set in the config file according to our rules.
    * @param weOwnDefaultSocket Whether Rancher Desktop has control over the default socket.
    * @param socketPath Path to the rancher-desktop specific docker socket. Darwin/Linux only.
    * @param kubernetesEndpoint Path to rancher-desktop Kubernetes endpoint.
    */
-  async ensureDockerConfig(weOwnDefaultSocket: boolean, socketPath?: string, kubernetesEndpoint?: string): Promise<void> {
+  async ensureDockerContextConfigured(weOwnDefaultSocket: boolean, socketPath?: string, kubernetesEndpoint?: string): Promise<void> {
     // read current config
     const currentConfig = await this.readDockerConfig();
 
-    console.log(`Read existing docker config: ${ JSON.stringify(currentConfig) }`);
     // Deep-copy the JSON object
     const newConfig = JSON.parse(JSON.stringify(currentConfig));
 
     // ensure docker context is set as we want
     const platform = os.platform();
 
     if ((platform === 'darwin' || platform === 'linux') && socketPath) {
-      await this.ensureDockerContext(socketPath, kubernetesEndpoint);
+      await this.ensureDockerContextFile(socketPath, kubernetesEndpoint);
     }
     newConfig.currentContext = await this.getDesiredDockerContext(weOwnDefaultSocket, currentConfig.currentContext);
 
-    // ensure we are using a valid credential helper
-    if (!newConfig.credsStore) {
-      newConfig.credsStore = this.getDefaultDockerCredsStore();
-    } else if (newConfig.credsStore === 'desktop' && !(await this.credHelperWorking(newConfig.credsStore))) {
-      newConfig.credsStore = this.getDefaultDockerCredsStore();
+    // write config if modified
+    if (JSON.stringify(newConfig) !== JSON.stringify(currentConfig)) {
+      await this.writeDockerConfig(newConfig);
     }
+  }
+
+  /**
+   * Ensures that the docker config file is configured with a valid credential helper.
+   */
+  async ensureCredHelperConfigured(): Promise<void> {
+    // read current config
+    const currentConfig = await this.readDockerConfig();
+
+    // Deep-copy the JSON object
+    const newConfig = JSON.parse(JSON.stringify(currentConfig));
+
+    // ensure we are using one of our preferred credential helpers
+    newConfig.credsStore = this.getCredsStoreFor(currentConfig.credsStore);
 
     // write config if modified
     if (JSON.stringify(newConfig) !== JSON.stringify(currentConfig)) {
```

Comment:
I disagree with this change. I had a working custom cred provider on my SUSE box, and now it gets replaced by `pass`, which hasn't been initialized, so it doesn't work.

Reply:
... and I now agree with the change. We'll write up instructions on installing and configuing `pass`.